modules/serverless-typescript-context-object-reference.adoc

```diff
@@ -27,12 +27,12 @@ export function handle(context: Context): string {
 }
 ----
 
-You can access the function by using the `kn func emit` command to invoke it:
+You can access the function by using the `kn func invoke` command:
 
 .Example command
 [source,terminal]
 ----
-$ kn func emit --sink 'http://example.function.com'
+$ kn func invoke --target 'http://example.function.com'
```

Comment:
The `--target` flag is not required, but it is shown in all of the examples in this document. I recommend removing it from at least one so that readers don't think it is required.

Reply:
Hmm, we have these four minor sections, each containing some command listings:
`Example commands specifying event parameters by using flags`
`Example commands specifying a file on disk that contains the event parameters`
`Example commands specifying a path to the function`
`Example commands specifying where the target function is deployed`
Of these, only the first and fourth have the `--target` flag. The first section plays the same role as a synopsis in manpages, the last section is specifically about running the command while specifying the target.
Are you then saying that specifying the target can be done without the `--target` flag?

Reply:
> The first section plays the same role as a synopsis in manpages, the last section is specifically about running the command while specifying the target.

That all looks good. But there are also changes in the JavaScript and TypeScript `context` object reference docs.

> Are you then saying that specifying the target can be done without the --target flag?

This is from `func help invoke`

```
	Invokes the Function by sending a test request to the currently running
	Function instance, either locally or remote.  If the Function is running
	both locally and remote, the local instance will be invoked.  This behavior
	can be manually overridden using the --target flag.
```

So when the function is running locally and the user runs `func invoke` with no arguments, it will be invoked locally. If it's running remotely on the cluster but not locally, it will be invoked on the cluster. If you want to alter this behavior, use the `--target` flag.

So in this file - the TypeScript reference, I see the following examples.

```
$ kn func invoke --target 'http://example.function.com'
```

```
$ kn func invoke --target 'http://example.function.com' --data '{"name": "tiger"}'
```

```
$ kn func invoke --target 'http://example.function.com' --data '{"hello": "world"}'
```

Note that they all use the `--target` flag.

In the javascript examples, I see the following examples.

```
$ kn func invoke --target 'http://example.function.com'
```

```
$ kn func invoke --target 'http://example.com?name=tiger'
```

```
$ kn func invoke -d '{"Hello": "world"}'
```

```
$ kn func invoke --target 'http://example.function.com'
```

I recommend removing `--target` from all of them, because if the function is running locally or remotely, it should "just work".

Reply:
@lance Ah, thanks for the explanation, now I got it. Made the requisite changes.

Could you please check what I say in the very last command example - that `--target local` with no running local function fails?